gobblin-restli/gobblin-throttling-service/gobblin-throttling-service-client/src/main/java/org/apache/gobblin/util/limiter/BatchedPermitsRequester.java

```diff
@@ -364,10 +393,22 @@ public void onSuccess(Response<PermitAllocation> result) {
       }
     }
 
+    public long elapsedTime() {
+      return System.currentTimeMillis() - this.startTime;
+    }
+
+    public synchronized void clearCallback() {
```

Comment:
Why this method is needed if there's only one on-the-fly request ( corresponding one callback) being existed ? Can't it be simply `clearSemaphore` instead of having a `callbackcleared` method for protection?

Reply:
We want to make sure a late callback and a timeout don't both release the semaphore.